examples/search_augmentation.ipynb

```diff
@@ -0,0 +1,472 @@
+    "## 1. Search\n",
+    "\n",
+    "Let's first generate a set of queries given a user question.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['San Francisco recent events',\n",
+       " 'news in San Francisco',\n",
+       " 'San Francisco happenings',\n",
+       " 'recent developments in San Francisco',\n",
+       " 'Happening in San Francisco today']"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# User asks a question\n",
+    "USER_QUESTION = \"What has happened recently in San Francisco?\"\n",
+    "\n",
+    "# Model generates a list of queries\n",
+    "PROMPT = f\"\"\"\n",
+    "Generate an array of search queries that are relevant to this question. \n",
+    "Use a variation of related keywords for the queries, trying to be as general as possible.\n",
+    "Include as many queries as you can think of, including and excluding terms. \n",
+    "For example, include queries like ['keyword_1 keyword_2', 'keyword_1', 'keyword_2']. \n",
+    "Be creative. The more queries you include, the more likely you are to find relevant results.\n",
+    "\n",
+    "User question: {USER_QUESTION}\n",
+    "\n",
+    "Format: {{\"queries\": [\"query_1\", \"query_2\", \"query_3\"]}}\n",
+    "Maximum 5 queries\n",
+    "\"\"\"\n",
+    "\n",
+    "queries = json_gpt(PROMPT)[\"queries\"]\n",
+    "\n",
+    "queries\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The queries look good, let's run the search!\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 5/5 [00:02<00:00,  2.30it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "def search_news(query: str):\n",
+    "    # get date 1 week ago\n",
+    "    one_week_ago = (date.today() - timedelta(weeks=1)).strftime(\"%Y-%m-%d\")\n",
+    "\n",
+    "    response = requests.get(\n",
+    "        \"https://newsapi.org/v2/everything\",\n",
+    "        params={\n",
+    "            \"q\": query,\n",
+    "            \"apiKey\": news_api_key,\n",
+    "            \"pageSize\": 50,\n",
+    "            \"sortBy\": \"relevancy\",\n",
+    "            \"from\": one_week_ago,\n",
+    "        },\n",
+    "    )\n",
+    "\n",
+    "    return response.json()\n",
+    "\n",
+    "\n",
+    "articles = []\n",
+    "\n",
+    "for query in tqdm(queries):\n",
+    "    articles = articles + search_news(query)[\"articles\"]\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of articles: 185\n",
+      "Top 5 articles: \n",
```

Comment:
I got a little confused here: are these the top 5 of query #1? did the prior step just concatenate our relevancy-ranked lists together?